src/utils/calculateUtils.ts

```diff
@@ -0,0 +1,46 @@
+import { Panel } from "../types/types";
+
+/**
+ * 개별 패널의 style 객체를 계산하는 함수
+ *
+ * @param panel - 위치와 크기 정보를 담은 패널 객체
+ * @param unitWidth - 1 grid column 당 픽셀 너비
+ * @param rowHeight - 1 grid row 당 픽셀 높이
+ * @returns React에서 사용 가능한 CSSProperties 객체
+ */
+interface calculatePanelStyleProps {
+  panel: Panel,
```

Comment:
타입을 엄격하게 정의해두면 나중에 유연성이 떨어진다고 느껴지는 때가 있습니다.
이럴 때 꼭 필요한 타입만 설정해두면 필수 속성만 일치하면 *정확하게* Panel이 아니어도 동작하는 유연한 코드가 됩니다.
이 정도 규모의 프로젝트에서 필수는 아니지만 알아두시라고 적어둡니다.

```suggestion
  dimensions: Pick<Panel, 'w' | 'h' | 'x' | 'y'>,
```

Reply:
네 참고하여 적용해보겠습니다!